week1/scary/444c43/Main.hs

```diff
@@ -0,0 +1,23 @@
+import System.IO
+import Data.Char
+import Data.List
+
+-- given a string
+--   convert the head to a number, if it's a letter
+--   subtract the offset and add the value of tail passed to self
+sumString :: String -> Int
+sumString []                    = 0
+sumString (x:xs)
+  | (x `elem` letters) == True  = (fromEnum (toLower x) - offset) + sumString xs
+  | (x `elem` letters) == False = 0 + sumString xs
+  where letters = ['a'..'z'] ++ ['A'..'Z']
+        offset  = 96
+
+-- create a new list with each element of a list whose summed total is 13
+parseList :: [String] -> [String]
+parseList xs = [ x | x <- xs, sumString x == 13]
```

Comment:
This is obviously fine, but another common variant is `parseList = filter ((== 13) . sumString)` and lots of experienced Haskellers write that way instead (so you should just be aware of it).

\<rant\>
My advice, however, is not to force yourself to mimic what experienced Haskellers do. Instead:

* when you're in a team, write as they write
* when you're on your own, write the way that seems most readable *to you*

A good experienced Haskeller will write code that is both readable and doesn't spell out things that pretty much everybody understands (like using general recursion instead of `map`, or something like that), but the only way to become such a Haskeller is to read lots of others' code. If you merely assume that all experienced Haskellers write good code, or that more complicated code is better, you'll end up writing _very_ bad code. 
\</rant\>